File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations (int [] nums, int k) {
        Arrays.sort (nums);
        List <Integer> b = new ArrayList <> ();
        for (int i = 0, j = 0, count = 0, x, y; ; ++count) {
            if (i < nums.length && (j >= b.size () || nums [i] <= b.get (j))) x = nums [i++];
            else x = b.get (j++);
            if (x >= k) return count;
            if (i < nums.length && (j >= b.size () || nums [i] <= b.get (j))) y = nums [i++];
            else y = b.get (j++);
            long t = 2L * x + y;
            b.add (t < k ? (int) t : k);
    }
}
}

File: 3332-minimum-operations-to-exceed-threshold-value-ii/3332-minimum-operations-to-exceed-threshold-value-ii.py
class Solution:
    def minOperations (self, nums : List [int], k : int) -> int:
        nums.sort ()
        b = []
        i = j = count = 0
        
        while True:
            if i < len (nums) and (j >= len (b) or nums [i] <= b [j]):
                x = nums [i]
                i += 1
            else:
                x = b [j]
                j += 1
            
            if x >= k:
                return count
            
            if i < len(nums) and (j >= len (b) or nums [i] <= b [j]):
                y = nums [i]
                i += 1
            else:
                y = b [j]
                j += 1
            
            b.append (2 * x + y)
            count += 1
        return -1